src/plugins/score-plugin-scenario/Scenario/Inspector/ObjectTree/SearchReplaceWidget.cpp

```diff
@@ -0,0 +1,327 @@
+          //          if(!must_add)
+          //            must_add = Process::hasMatchingText(root, stxt, listCache, messagesCache);
+          // FIXME look into state processes?
+
+          // Try to add if the searched text is in the name of the state
+          if(must_add)
+            m_matches.push_back(obj);
+          //          else
+          //            return;
+          //            add_if_contains(*state, stxt, sel);
+        }
+        else if(auto event = qobject_cast<const EventModel*>(obj))
+        {
+          if(State::findAddressInExpression(event->condition(), m_oldAddress))
+          {
+            m_matches.push_back(obj);
+            sel.append(*event);
+          }
+          //            add_if_contains(*event, stxt, sel);
+        }
+        else if(auto ts = qobject_cast<const TimeSyncModel*>(obj))
+        {
+
+          if(State::findAddressInExpression(ts->expression(), m_oldAddress))
+            m_matches.push_back(obj);
+          sel.append(*ts);
+          //            add_if_contains(*ts, stxt, sel);
+        }
+        else if(auto cmt = qobject_cast<const CommentBlockModel*>(obj))
+        { /*
+           if(State::findAddressInExpression(expr,addresses[0].address))
+                    m_matches.push_back(obj);*/
+          //            add_if_contains(*cmt, stxt, sel);
+        }
+        else if(auto interval = qobject_cast<const IntervalModel*>(obj))
+        {
+          for(auto& proc : interval->processes)
+          {
+            selectProcessPortsWithAddress(addresses, proc, sel);
+            for(Process::Inlet* inlet : proc.inlets())
+            {
+              if(sel.contains(inlet))
+              {
+                m_matches.push_back(inlet);
+              }
+            }
+            for(Process::Outlet* outlet : proc.outlets())
+            {
+              if(sel.contains(outlet))
+              {
+                m_matches.push_back(outlet);
+              }
+            }
+            //                        add_if_contains(proc, stxt, sel);
+            sel.append(proc);
+          }
+
+          //          add_if_contains(*interval, stxt, sel);
+        }
+      }
+    }
+    score::SelectionDispatcher d{doc->context().selectionStack};
+    d.select(sel);
+  }
+}
+
+void SearchReplaceWidget::replace()
+{
+  auto& ctx = m_ctx.documents.currentDocument()->context();
+  MacroCommandDispatcher<Command::ReplaceAddresses> disp{ctx.commandStack};
+  if(!m_newAddress.isSet())
+  {
+    return;
+  }
+  else
+  {
+    for(auto* obj : m_matches)
+    {
+      if(auto state = qobject_cast<const StateModel*>(obj))
+      {
+        auto& root = state->messages().rootNode();
+        //TODO maybe implement a better ver of RenameAddressInState ?
+        disp.submit(new Command::RenameAddressInState(
+            *state, State::AddressAccessor{m_oldAddress},
+            State::AddressAccessorHead{m_newAddress.path.back(), {}}));
```

Comment:
bug here: if the address is like "foo:/" then path will be empty and this will crash.
for now you can just add a `if(!m_newAddress.path.isEmpty())` guard in the beginning